src/main/scala/com/gu/subscriptions/cas/service/ZuoraSubscriptionService.scala

```diff
@@ -93,8 +95,10 @@ object ZuoraClient extends ZuoraApi with ZuoraClient {
   lazy val authTask = ScheduledTask(s"Zuora ${apiConfig.envName} auth", Authentication("", ""), 0.seconds, 30.minutes)(
     request(Login(apiConfig)))
 
-  def queryForSubscription(subscriptionId: String): Future[Zuora.Subscription] =
-    queryOne[Zuora.Subscription](s"Name='$subscriptionId'")
+  def queryForSubscription(subscriptionName: String): Future[Zuora.Subscription] =
```

Comment:
Is this querying the subscription versions? How many records are there normally in this Zuora table (does a new one get created whenever a subscription is amended)? Maybe it is the case of monitoring the execution time of this query, as performance might degrade dramatically if for some reasons we end up with a lot of versions